lib/facility_factory.rb

```diff
@@ -0,0 +1,33 @@
+class Facility_Factory
+    def create_facilities(sources)
+      sources.map do |source|
+          facility = Facility.new({
+            name: source[:dmv_office] || source[:office_name] || source[:name],
+            phone: source[:phone] || source[:public_phone_number],
+            address: address(source)})
+      end
+    end
+
+    def address(source)
+        address_parsed = {
+            address_ln1: [:address_li, :street_address_line_1, :address1],
+            address_ln2: [:address__1, :street_address_line_2],
+            address_ln3: [:city],
+            address_ln4: [:state],
+            address_ln5: [:zip, :zip_code, :zipcode]
+                }
+        address_bits = []
```

Comment:
Maybe a better variable name here, but this works.

Reply:
variable name could have been address_parts